src/Analyzers/Core/CodeFixes/UpgradeProject/AbstractUpgradeProjectCodeFixProvider.cs

```diff
@@ -95,6 +94,32 @@ private bool CanUpgrade(Project project, string language, string version)
             => project.Language == language && IsUpgrade(project, version);
     }
 
+#if CODE_STYLE
+
+    internal class ProjectOptionsChangeAction : CodeAction
```

Comment:
I think you can get rid of `ProjectOptionsChangeAction` completely, both in CodeStyle and Features contexts of this file. It doesn't seem this type is serving any purpose or passing any special arguments that the public API `CodeAction.Create` wouldn't be able to support.